src/main/java/gyro/aws/autoscaling/AutoScalingPlanResource.java

```diff
@@ -0,0 +1,286 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.aws.autoscaling;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import software.amazon.awssdk.services.autoscalingplans.AutoScalingPlansClient;
+import software.amazon.awssdk.services.autoscalingplans.model.CreateScalingPlanResponse;
+import software.amazon.awssdk.services.autoscalingplans.model.DescribeScalingPlansResponse;
+import software.amazon.awssdk.services.autoscalingplans.model.ScalingPlan;
+import software.amazon.awssdk.services.autoscalingplans.model.ScalingPlanStatusCode;
+
+/**
+ * Creates a Scaling plan.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    aws::autoscaling-plan autoscaling-plan-example
+ *        scaling-plan-name: "scaling-plan-example"
+ *        scaling-plan-version: 1.0
+ *        status-code: "Active"
+ *
+ *        application-source
+ *            cloud-formation-stack-arn: "arn:aws:cloudformation:us-east-2:242040583208:stack/example-stack/e95a0400-192f-11eb-8eb9-0a36006b772c"
+ *        end
+ *
+ *        scaling-instructions
+ *            max-capacity: 100
+ *            min-capacity: 0
+ *            service-namespace: "autoscaling"
+ *            resource-id: "autoScalingGroup/test-autoscaling-group"
+ *            scalable-dimension: "autoscaling:autoScalingGroup:DesiredCapacity"
+ *
+ *            target-tracking-configuration
+ *                target-value: 2.0
+ *
+ *                customized-scaling-metric-specification
+ *                    dimensions
+ *                        name: "dimension-test"
+ *                        value: "dimension-test-value"
+ *                    end
+ *
+ *                    metric-name: "metric-name-test"
+ *                    namespace: "namespace"
+ *                    statistic: "Sum"
+ *                end
+ *            end
+ *        end
+ *    end
+ *
+ */
+@Type("autoscaling-plan")
+public class AutoScalingPlanResource extends AwsResource implements Copyable<ScalingPlan> {
+
+    private AutoScalingApplicationSource applicationSource;
+    private String creationTime;
+    private List<AutoScalingScalingInstruction> scalingInstructions;
+    private String scalingPlanName;
```

Comment:
Change the name across all getters setters and examples bot docs and .gyro files
```suggestion
    private String Name;
```